lienzo-core/src/main/java/com/ait/lienzo/client/core/types/JsLienzo.java

```diff
@@ -0,0 +1,235 @@
+            animations = new JsLienzoAnimations();
+        }
+        return animations;
+    }
+
+    public JsLienzoLogger log() {
+        if (null == logger) {
+            logger = new JsLienzoLogger(this);
+        }
+        return logger;
+    }
+
+    public int getPanelOffsetLeft() {
+        int result = panel.getElement().offsetLeft;
+        return result;
+    }
+
+    public int getPanelOffsetTop() {
+        int result = panel.getElement().offsetTop;
+        return result;
+    }
+
+    public void add(IPrimitive<?> shape) {
+        getLayer().add(shape);
+    }
+
+    public void draw() {
+        getLayer().draw();
+    }
+
+    public IPrimitive<?> getShape(String id) {
+        return getShapeInContainer(id, getLayer());
+    }
+
+    @SuppressWarnings("all")
+    private static IPrimitive<?> getShapeInContainer(String id, ContainerNode parent) {
+        NFastArrayList<IPrimitive<?>> shapes = parent.getChildNodes();
+        if (null != shapes) {
+            for (IPrimitive<?> shape : shapes.asList()) {
+                String shapeID = shape.getID();
+                if (id.equals(shapeID)) {
+                    return shape;
+                }
+                if (shape instanceof ContainerNode) {
+                    IPrimitive<?> shape1 = getShapeInContainer(id, (ContainerNode) shape);
+                    if (null != shape1) {
+                        return shape1;
+                    }
+                }
+            }
+        }
+        return null;
+    }
+
+    public WiresManager getWiresManager() {
+        return WiresManager.get(getLayer());
+    }
+
+    public JsWiresShape getWiresShape(String id) {
+        WiresShape[] shapes = getWiresManager().getShapes();
+        for (WiresShape shape : shapes) {
+            if (id.equals(shape.getID())) {
+                final JsWiresShape jsWiresShape = new JsWiresShape(shape);
+                return jsWiresShape;
+            }
+        }
+        return null;
+    }
+
+    public String getBackgroundColor(String UUID) {
+        JsWiresShape shape = getWiresShape(UUID);
+        if (shape == null) {
+            return null;
+        }
+        return shape.getBackgroundColor();
+    }
+
+    public void setBackgroundColor(String UUID, String backgroundColor) {
+        JsWiresShape shape = getWiresShape(UUID);
+        if (shape == null) {
+            return;
+        }
+        shape.setBackgroundColor(backgroundColor);
+    }
+
+    public String getBorderColor(String UUID) {
+        JsWiresShape shape = getWiresShape(UUID);
+        if (shape == null) {
+            return null;
+        }
+        return shape.getBorderColor();
+    }
+
+    public void setBorderColor(String UUID, String borderColor) {
+        JsWiresShape shape = getWiresShape(UUID);
+        if (shape == null) {
+            return;
+        }
+        shape.setBorderColor(borderColor);
+    }
+
+    public NFastArrayList<Double> getLocation(String UUID) {
```

Comment:
Calling this without `UUID` e.g. `getLocation()` throws NPE. Could this be addressed or is this expected? Same goes for the other methods.

Reply:
Hi @domhanak will fix